crates/bounded_planet_core/src/networking/streams.rs

```diff
@@ -83,7 +83,7 @@ impl<TSession: Session> BoundedPlanetRecvStream<TSession> {
         let length_prefix = u32::from_be_bytes(length_prefix_buf);
 
         // Read that many bytes
-        let mut data = Vec::<u8>::with_capacity(length_prefix as usize);
+        let mut data = vec![0; length_prefix as usize];
```

Comment:
Just to make sure this was intended, this vec is now zero'd as well as of the given size.